opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -232,119 +261,151 @@ message MetricDescriptor {
+  // the point was aggregated with resepect to time (Temporality), and
+  // whether it was computed synchronously, and when the Structure is
+  // ADDING, whether the captured Sum is also monotonic.
+  //
+  // Kind names are generated from valid combinations of KindElement
+  // by joining the effective KindElements using underscores.  There
+  // are:
+  //
+  // - 3 possibilities for Temporality
+  // - 2 possibilities for Synchronous/Asynchronous
+  // - 3 possibilities for Structure/Monotonicity: Grouping, Adding
+  //   Monotonic, and Adding (not monotonic)
+  //
+  // This makes 18 valid values.
+  enum Kind {
+    // INVALID_KIND is the default Kind, it MUST not be used.
+    INVALID_KIND = 0;
+
+    // The following codes are generated by a program.
+
+    ADDING_MONOTONIC_INSTANTANEOUS_SYNCHRONOUS  = 0x71;
+    ADDING_MONOTONIC_INSTANTANEOUS_ASYNCHRONOUS = 0x31;
+    ADDING_MONOTONIC_CUMULATIVE_SYNCHRONOUS     = 0x72;
+    ADDING_MONOTONIC_CUMULATIVE_ASYNCHRONOUS    = 0x32;
+    ADDING_MONOTONIC_DELTA_SYNCHRONOUS          = 0x74;
+    ADDING_MONOTONIC_DELTA_ASYNCHRONOUS         = 0x34;
+
+    ADDING_INSTANTANEOUS_SYNCHRONOUS            = 0x51;
+    ADDING_INSTANTANEOUS_ASYNCHRONOUS           = 0x11;
+    ADDING_CUMULATIVE_SYNCHRONOUS               = 0x52;
+    ADDING_CUMULATIVE_ASYNCHRONOUS              = 0x12;
+    ADDING_DELTA_SYNCHRONOUS                    = 0x54;
+    ADDING_DELTA_ASYNCHRONOUS                   = 0x14;
+
+    GROUPING_INSTANTANEOUS_SYNCHRONOUS          = 0x49;
+    GROUPING_INSTANTANEOUS_ASYNCHRONOUS         = 0x9;
+    GROUPING_CUMULATIVE_SYNCHRONOUS             = 0x4a;
+    GROUPING_CUMULATIVE_ASYNCHRONOUS            = 0xa;
+    GROUPING_DELTA_SYNCHRONOUS                  = 0x4c;
+    GROUPING_DELTA_ASYNCHRONOUS                 = 0xc;
   }
 
-  // temporality is the Temporality of values this metric has.
-  Temporality temporality = 5;
+  // Kind describes properties of the Metric that are necessary to
+  // interpret the data and/or describe how it was produced.
+  Kind kind = 5;
 }
 
-// Int64DataPoint is a single data point in a timeseries that describes the time-varying
-// values of a int64 metric.
-message Int64DataPoint {
+// DataPoint is a point of raw or aggregate metric data, generally containing
+// the labels, one or two timestamps, and a single value field.  The value field
+// that is used is determined by the descriptor ValueType.
+message DataPoint {
   // The set of labels that uniquely identify this timeseries.
   repeated opentelemetry.proto.common.v1.StringKeyValue labels = 1;
 
-  // start_time_unix_nano is the time when the cumulative value was reset to zero.
-  // This is used for Counter type only. For Gauge the value is not specified and
-  // defaults to 0.
+  // start_time_unix_nano is the start of the interval for which a CUMULATIVE or
+  // DELTA kind of metric was computed.
+  // reset.
+  //
+  // This must not be set for INSTANTANEOUS metric kinds.
```

Comment:
Should this be changed to "is ignored for INSTANTANEOUS"?

If not, should we make this normative?

```suggestion
  // This MUST not be set for INSTANTANEOUS metric kinds.
```

or

```suggestion
  // This SHOULD not be set for INSTANTANEOUS metric kinds.
```

Reply:
I wrote:
```
  // This field is ignored for INSTANTANEOUS metric kinds and SHOULD not be set.
```